code/ship/ship.cpp

```diff
@@ -11952,6 +11953,16 @@ int ship_fire_secondary( object *obj, int allow_swarm )
 
 		int start_slot, end_slot;
```

Comment:
Put these either at the top of the block (where `num_slots` is) or below your new addition (because the next block is the code it was next to originally, and it should still be associated with it).

Reply:
If moving down, ought num_slots declaration and assignment be moved down below my addition as well? It's not used until that section.

Reply:
If you want to, but `num_slots` is not part of the scope of this PR.  What I wanted to point out was that, in adding the new if() block, you separated two pieces of code that were intended to be associated together.  Be careful about that.

Reply:
Reasonable, I'll keep my changes local then.